bin/network-monitor/src/faucet.rs

```diff
@@ -0,0 +1,240 @@
+//! Faucet testing functionality.
+//!
+//! This module contains the logic for periodically testing faucet functionality
+//! by requesting proof-of-work challenges, solving them, and submitting token requests.
+
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+use anyhow::Context;
+use miden_objects::account::AccountId;
+use miden_objects::testing::account_id::ACCOUNT_ID_SENDER;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+use tokio::sync::watch;
+use tokio::time::MissedTickBehavior;
+use tracing::{debug, info, instrument, warn};
+use url::Url;
+
+use crate::COMPONENT;
+use crate::status::{ServiceDetails, ServiceStatus, Status};
+
+// FAUCET TEST TYPES
+// ================================================================================================
+
+/// Details of a faucet test.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct FaucetTestDetails {
+    pub test_duration_ms: u64,
+    pub success_count: u64,
+    pub failure_count: u64,
+    pub last_tx_id: Option<String>,
+    pub last_note_id: Option<String>,
+    pub challenge_difficulty: Option<u64>,
+}
+
+/// Response from the faucet's `/pow` endpoint.
+#[derive(Debug, Deserialize)]
+struct PowChallengeResponse {
+    challenge: String,
+    target: u64,
+    #[allow(dead_code)] // Timestamp is part of API response but not used
+    timestamp: u64,
+}
+
+/// Response from the faucet's `/get_tokens` endpoint.
+#[derive(Debug, Deserialize)]
+struct GetTokensResponse {
+    tx_id: String,
+    note_id: String,
+}
+
+// FAUCET TEST TASK
+// ================================================================================================
+
+/// Runs a task that continuously tests faucet functionality and updates a watch channel.
```

Comment:
Upon a quick look, I couldn't find how frequently we make the request to the faucet. We probably don't need to do this too frequently. Maybe once every 30 seconds is enough? Either way, I'd document this somewhere (unless it is already documented and I missed it).

Reply:
In the last commit, i create constants for the before mentioned values, and included the frequency. I set it to 2 minutes to match the frequency of the tests and make sure that we are not disturbing the faucet's challenge difficulty. Also, in a follow up PR i'll add `clap` to the binary since configuration grew too large and management thorugh clap is easer, i plan to add this interval as a configuration option in that PR.